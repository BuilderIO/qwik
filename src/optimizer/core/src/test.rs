extern crate insta;
use super::*;

#[test]
fn example_1() {
    test_input(
        "test.tsx",
        r#"
const Header = qComponent({
  "onMount": qHook(() => { console.log("mount") }),
  onRender: qHook(() => {
    return (
      <div onClick={qHook((ctx) => console.log(ctx))}/>
    );
  })
});
    "#,
    )
}

#[test]
fn example_2() {
    test_input(
        "test.tsx",
        r#"
export const Header = qComponent({
  "onMount": qHook(() => { console.log("mount") }),
  onRender: qHook(() => {
    return (
      <div onClick={qHook((ctx) => console.log(ctx))}/>
    );
  })
});
    "#,
    )
}

#[test]
fn example_3() {
    test_input(
        "test.tsx",
        r#"
export const App = () => {
    const Header = qComponent({
        "onMount": qHook(() => { console.log("mount") }),
        onRender: qHook(() => {
            return (
            <div onClick={qHook((ctx) => console.log(ctx))}/>
            );
        })
    });
    return Header;
});
    "#,
    )
}

#[test]
fn optimize_fixture_1() {
    test_fixture("");
}

fn test_fixture(folder: &str) {
    let res = transform_workdir(&FSConfig {
        project_root: folder.to_string(),
        source_maps: true,
        minify: false,
        transpile: false,
    });
    insta::assert_debug_snapshot!(res);
}

fn test_input(filename: &str, code: &str) {
    let res = transform(Config {
        code: code.as_bytes().to_vec(),
        filename: filename.to_string(),
        source_maps: true,
        minify: false,
        transpile: false,
    });
    match res {
        Ok(v) => {
            let s = v.to_string();
            insta::assert_display_snapshot!(if let Some(code) = s.code { code } else { "".to_string() });
            insta::assert_display_snapshot!(if let Some(map) = s.map { map } else { "".to_string() });
            insta::assert_debug_snapshot!(v.diagnostics);
        }
        Err(err) => {
            insta::assert_display_snapshot!(err);
        }
    }
}
