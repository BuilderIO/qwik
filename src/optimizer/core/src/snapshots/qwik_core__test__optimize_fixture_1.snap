---
source: src/test.rs
expression: res

---
Err(
    Os {
        code: 2,
        kind: NotFound,
        message: "No such file or directory",
    },
)
